core/src/main/scala/org/ensime/indexer/ClassfileIndexer.scala

```diff
@@ -187,6 +187,16 @@ trait ClassfileIndexer {
         case member => FieldName(ClassName.fromInternal(owner), member)
       }
 
+    private def extractClassNames(className: String): Seq[String] = {
```

Comment:
Let's say we have 
```
package org.foo

object Foo {
  class Inner {} 
}
```
 in Foo.scala and
```
object Bar extends App {
  val a = new Foo.Inner()
}
```
 in Bar.scala. The way we had it before is `val a` only had a reference to `org.foo.Foo$Inner`. Now suppose we want to rename `object Foo` we lookup the usages and `val a` won't come up, because it "uses" `org.foo.Foo$Inner` not `org.foo.Foo$`. In short we fail to detect any uses of outer classes when their inner classes are called, which is obviously faulty. This is sort of a hacky way to achieve what we want here (i.e. extract all possible names of outer classes).